it seems that meta::transpose does not work

using ListOfLists = meta::list< meta::list<char>,  meta::list<std::string, int>,	    meta::list<unsigned, long> >;

  using tp_t = meta::transpose<ListOfLists>;

when the above code get compiled, the following errors are generated.

`/Users/Michael_Z/open_source/meta/include/meta/meta.hpp:140:9: error: no type named
      'type' in 'meta::v1::detail::fold_<meta::v1::list<meta::v1::list<char>,
      meta::v1::list<std::__1::basic_string<char>, int>, meta::v1::list<unsigned int,
      long> >, meta::v1::id<meta::v1::list<meta::v1::list<> > >,
      meta::v1::bind_back<meta::v1::quote<transform>, meta::v1::quote<push_back> > >'
        using _t = typename T::type;
        ^~~~~
/Users/Michael_Z/open_source/meta/include/meta/meta.hpp:1303:9: note: in
      instantiation of template type alias '_t' requested here
        using fold = _t<detail::fold_<List, id<State>, Fun>>;
        ^
/Users/Michael_Z/open_source/meta/include/meta/meta.hpp:2452:9: note: in
      instantiation of template type alias 'fold' requested here
        using transpose = fold<ListOfLists,
        ^
test_meta.cpp:12:22: note: in instantiation of template type alias 'transpose'
      requested here
  using tp_t = meta::transpose<listOfLists>;
                     ^
1 error generated.
`

anyone can figure it out ? thanks.

--- a/include/meta/meta.hpp
+++ b/include/meta/meta.hpp
@@ -2839,10 +2839,11 @@ namespace meta
 
     ///////////////////////////////////////////////////////////////////////////////////////////
     // transpose
-    /// Given a list of lists of types \p ListOfLists, transpose the elements from the lists.
+    /// Given a list of lists of the same number of types \p ListOfLists, transpose the
+    /// elements from the lists.
     /// \par Complexity
-    /// \f$ O(N \times M) \f$, where \f$ N \f$ is the size of the outer list, and
-    /// \f$ M \f$ is the size of the inner lists.
+    /// \f$ O(N \times M) \f$, where \f$ N \f$ is the size of the outer list, and \f$ M \f$ is
+    /// the size of all of the inner lists.
     /// \ingroup transformation
     template <META_TYPE_CONSTRAINT(List) ListOfLists>
     using transpose = fold<ListOfLists, repeat_n<size<front<ListOfLists>>, list<>>,

